lib/crypto.js

```diff
@@ -652,6 +672,23 @@ function filterDuplicates(names) {
   }).sort();
 }
 
+// timingSafeEqual reuses the timing-safe Hmac.equals function (see above) for
+// comparison of non-hash inputs, like capability URLs or authentication tokens.
+// Note that the strength of the double-HMAC construction does not depend
+// on the unpredictability of the key. Since randomBytes can potentially
+// throw an exception, we use Math.random, which sacrifices nothing in timing
+// safety and benefits from never throwing an exception.
+// See discussion at
+// https://github.com/nodejs/node-v0.x-archive/issues/8560#issuecomment-60145816
+exports.timingSafeEqual = function(a, b) {
+  var key = new Buffer(32);
+  for (var i = 0; i < 32; i += 4) {
+    key.writeFloatBE(Math.random(), i);
```

Comment:
Well, you could make it sync/async just like `randomBytes()`. That way an error is thrown for sync or it's passed as the `err` argument for the async callback. Then use the async variation if you really don't want to wrap `timingSafeEqual()` in a try-catch.